models/AmazonS3.cfc

```diff
@@ -1015,27 +1046,97 @@ component accessors="true" singleton {
 		return "#HTTPPrefix##hostname##variables.signatureUtil.buildCanonicalURI( arguments.uri )#?#sigData.canonicalQueryString#&X-Amz-Signature=#sigData.signature#";
 	}
 
+	struct function createMultiPartUpload(
+		required string bucketName,
+		required string uri,
+		string  contentType         = "",
+		string  contentEncoding     = "",
+		string contentDisposition   = "",
+		numeric HTTPTimeout         = variables.defaultTimeout,
+		string  cacheControl        = variables.defaultCacheControl,
+		string  expires             = 120,
+		any     acl                 = variables.defaultACL,
+		struct  metaHeaders         = {},
+		string  md5                 = variables.autoMD5,
+		string  storageClass        = variables.defaultStorageClass,
+		string  encryptionAlgorithm = variables.defaultEncryptionAlgorithm,
+		string  encryptionKey       = variables.defaultEncryptionKey
+	){
+		requireBucketName( arguments.bucketName );
+		var headers = createMetaHeaders( arguments.metaHeaders );
+		applyACLHeaders( headers, arguments.acl );
+		applyEncryptionHeaders( headers, arguments );
+
+		if ( len( arguments.storageClass ) ) {
+			headers[ "x-amz-storage-class" ] = arguments.storageClass;
+		}
+
+		if ( arguments.contentType == "auto" ) {
```

Comment:
If a contentType is not passed into createMultiPartUpload(), shouldn't this automatically determine the mime type?

```
if ( arguments.contentType == "auto" || !len( arguments.contentType ) ) {
```

Reply:
No, because then AWS will determine the mime type of what it is accepting.  That way, if an unknown mime type is passed, AWS will handle detection